common/luaclass.c

```diff
@@ -300,6 +300,27 @@ luaA_class_connect_signal_from_stack(lua_State *L, lua_class_t *lua_class,
                                      const char *name, int ud)
 {
     luaA_checkfunction(L, ud);
+
+    /* Duplicate the function in the stack */
+    lua_pushvalue(L, ud);
+    luaA_checkfunction(L, -1);
+    luaA_checkfunction(L, ud);
+
+    /* Create a new signal to notify there is a global connection. */
+    char buf[128];
+    a_strcpy(buf, 128, name);
+    a_strcat(buf, 128, "::connected");
```

Comment:
Can we get rid of the 128 limit?

Untested:
```C
#define CONNECTED_SUFFIX "::connected"
char *buf = p_alloca(char, a_strlen(name) + a_strlen(CONNECTED_SUFFIX) + 1);
sprintf(buf, "%s%s", name, CONNECTED_SUFFIX);
#undef CONNECTED_SUFFIX
```

Reply:
I went for growing a static buffer. Might not be very clean when quitting Awesome but at least it doesn't do 1k small allocs at startup (yes, my config is big, I know).

Reply:
`p_alloca` allocates stack space, not heap space...?

Reply:
Is this like "elv13 please just use p_alloca, vla are cool" or is it a generic comment?

Reply:
I'm fine with `p_alloca` and `p_alloca` does not use a variable length array. So yes, they are cool. Also, `alloca` is a lot cheaper than `malloc` (it's just subtracting a number from the stack pointer to reserve some stack space), so I didn't understand the comment about the 1k small allocs. Also, I didn't look at that static buffer patch before now and I think `p_alloca` is simpler than that.

Reply:
done